tslearn/shapelets.py

```diff
@@ -195,21 +210,25 @@ class ShapeletModel(BaseEstimator, ClassifierMixin):
 
     Parameters
     ----------
-    n_shapelets_per_size: dict
+    n_shapelets_per_size: dict (default: None)
         Dictionary giving, for each shapelet size (key),
         the number of such shapelets to be trained (value)
-    max_iter: int (default: 1000)
+    max_iter: int (default: 100)
         Number of training epochs.
     batch_size: int (default:256)
         Batch size to be used.
-    verbose_level: {0, 1, 2} (default: 2)
+    verbose_level: {0, 1, 2} (default: 0)
```

Comment:
I agree, this will cause backward compatibility issues though (code with verbose_level will no longer work).

Reply:
If we want to make it clean, we could add a deprecation warning (see this [example](https://github.com/scikit-learn/scikit-learn/blob/a83c8311dfdbf74dea584d45c6f254bc8171054d/sklearn/model_selection/_search.py#L796-L800) in scikit-learn).

tslearn 0.2.0 will probably be a big update, so it would not mind breaking backward compatibility for code improvement. Let's wait for Romain's point of view.

Reply:
On this point I would say:
- we should change to `verbose`
- we should keep the `verbose_level` option active for some time before removing it (and print a deprecationwarning, and add a note in the docs that the option will be deprecated in a future version -- let's say 0.2.3 e.g.)

Reply:
Starting using [semantic versioning](https://semver.org) could be a good idea. A patch version is for patching only, and deprecations usually occur in minor or major versions.

Reply:
From your link:

Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

So we're ok no?

Reply:
scikit-learn is still in major version zero for instance, but patch versions are used for patches and minor versions are used for new features and deprecations (two minor versions for deprecation warnings).

I think that it would be a plus to follow semantic versioning, it’s a good practice.

Reply:
Ok for me!

So we could use the same strategy as sklearn starting from now and take care that we use semantic versioning starting from now.

That means deprecation warnings should state that the feature will be removed in 0.4